### PR TITLE
feat: just enough //extensions to load a simple devtools extension

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
Depends on #19512.

Ref #19447

This adds just enough infrastructure to get a simple devtools extension loading.

Tested to be working:
- the `"devtools_page"` manifest key
- the `chrome.devtools.panels.create` API
- loading JS from both the devtools page and the panel

brown-jar-seem

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

(//extensions still not turned on)

### CODE REVIEW

**Comment by nornagon** [comment]
@jkleinsc good thing you asked! found a couple bugs. will work on it more tomorrow.

**Comment by nornagon** [comment]
@jkleinsc I (eventually) made a test :) I hope I fixed whatever bug it was that I was thinking about, because I've long since forgotten what I encountered. Test works though.

File: shell/browser/atom_browser_client.cc [hunk 1]
```diff
@@ -988,6 +1007,7 @@ bool AtomBrowserClient::WillCreateURLLoaderFactory(
   v8::Isolate* isolate = v8::Isolate::GetCurrent();
   api::Protocol* protocol =
       api::Protocol::FromWrappedClass(isolate, browser_context);
+  // if (!protocol) return false;
```
Comment (nornagon) [inline shell/browser/atom_browser_client.cc]:
```suggestion
```

File: spec-main/extensions-spec.ts [hunk 2]
```diff
@@ -138,6 +138,45 @@ ifdescribe(process.electronBinding('features').isExtensionsEnabled())('chrome ex
...
+            showPanelTimeoutId = setTimeout(show, 100)
+          })
+        }
+        showPanelTimeoutId = setTimeout(show, 100)
+      })
+    }
+
+    it('loads a devtools extension', async () => {
+      const customSession = session.fromPartition(`persist:${require('uuid').v4()}`);
+      (customSession as any).loadChromeExtension(path.join(fixtures, 'extensions', 'devtools-extension'))
```
Comment (jkleinsc) [inline spec-main/extensions-spec.ts]: This needs to be updated to match the changes made in #21587
```suggestion
      (customSession as any).loadExtension(path.join(fixtures, 'extensions', 'devtools-extension'))
```

**Pull request merged**